test/unit/org/apache/cassandra/tools/ToolRunner.java

```diff
@@ -75,13 +77,6 @@ public ToolRunner(List<String> args, boolean runOutOfProcess)
         this.runOutOfProcess = runOutOfProcess;
     }
 
-    public ToolRunner withStdin(InputStream stdin, boolean autoClose)
```

Comment:
Why kill stdin?

Reply:
it was dead code and harder to reason how to deal with.  There are two patterns I see with stdin: one time input, and interactive; the interactive case seems hard to deal with if creating a thread to try to copy bytes.  Normally I see stdin exposed so the test can control when bytes are added, so would expect something like

```class ToolRunner {
...
OutputStream getStdin()
...```. and then the test can append to stdin when desired.

Reply:
Then again, we don't stream stdout so the interactive case isn't possible currently (do we even have that with any tool?)

Reply:
I will add it back for now.

Reply:
I added stdin back but mostly had to redo the logic.  Since we don't control stdin it might not be closed when the process exists which would cause threads to leak (as copy would hang forever); to work around this I now attempt to detect when the process exists and run callbacks (only used to close stdin).